nodes/config/ui_base.js

```diff
@@ -1130,7 +1131,17 @@ module.exports = function (RED) {
         const host = RED.settings.uiHost
         const port = RED.settings.uiPort
         const httpAdminRoot = RED.settings.httpAdminRoot
-        const url = 'http://' + (`${host}:${port}/${httpAdminRoot}flows`).replace('//', '/')
+        let scheme = 'http://'
+        let httpsAgent
+        if (RED.settings.https) {
+            const https = (typeof RED.settings.https === 'function' ? RED.settings.https() : RED.settings.https) || {}
```

Comment:
If `RED.settings.https` is a function, then it *might* return a promise that needs to be resolved first. Here's how the core handles it:

https://github.com/node-red/node-red/blob/master/packages/node_modules/node-red/red.js#L243

Reply:
thanks Nick. resolved in [1771bbe](https://github.com/FlowFuse/node-red-dashboard/pull/1488/commits/1771bbe77e72a8f5cb52fa0e3abaf700413d3e9c)